definitionsconsistencycheck.cmake

```diff
@@ -1,11 +1,11 @@
 get_directory_property( DirDefs COMPILE_DEFINITIONS )
 
 # Reset the definition file
-file(WRITE cmake.definitions "")
+file(WRITE $ENV{__IntermediatesDir}/cmake.definitions "")
```

Comment:
Could you please use ${CMAKE_CURRENT_BINARY_DIR} instead of the $ENV{__IntermediatesDir}? I would prefer to not to use env variables in the cmake scripts to communicate information between the build.sh and cmake scripts. 
I know we have it at two or three other places, but I'd like to eventually fix it there as well.